lib/percy/capybara/loaders/filesystem_loader.rb

```diff
@@ -0,0 +1,50 @@
+require 'percy/capybara/loaders/base_loader'
+require 'digest'
+require 'find'
+
+module Percy
+  module Capybara
+    module Loaders
+
+      # Resource loader that looks for resources in the specified folder.
+      class FilesystemLoader < BaseLoader
+        SKIP_RESOURCE_EXTENSIONS = [
+          '.map',  # Ignore source maps.
+          '.gz',  # Ignore gzipped files.
+        ]
+        MAX_FILESIZE_BYTES = 15 * 1024**2  # 15 MB.
+
+        def initialize(options = {})
+          # @asset_path should point to a _compiled_ static assets directory, not source assets.
+          @asset_path = options[:asset_path]
+          super
+        end
+
+        def snapshot_resources
+          [root_html_resource]
+        end
+
+        def build_resources
+          resources = []
+          Find.find(@asset_path).each do |path|
+            # Skip directories.
+            next if !FileTest.file?(path)
+            # Skip certain extensions.
+            next if SKIP_RESOURCE_EXTENSIONS.include?(File.extname(path))
+            # Skip large files, these are hopefully downloads and not used in page rendering.
+            next if File.size(path) > MAX_FILESIZE_BYTES
+
+            # Strip the public_path from the beginning of the resource_url.
+            # This assumes that everything in the public_path directory is served at the root
+            # of the app.
+            resource_url = path.sub(@asset_path, '')
```

Comment:
Per discussion, a common need here is to support production asset URLs that are different from filesystem paths (ie. I compile into "./dist/images/foo.png"  but in my dev/prod server that gets served at "/assets/images/foo.png").

This could be as simple as adding a `base_url` option.